editor/plugins/shader_editor_plugin.cpp

```diff
@@ -1284,12 +1288,15 @@ void ShaderEditorPlugin::edit(Object *p_object) {
 			}
 		}
 		es.shader = Ref<Shader>(s);
+#ifdef MODULE_VISUAL_SHADER_ENABLED
 		Ref<VisualShader> vs = es.shader;
 		if (vs.is_valid()) {
 			es.visual_shader_editor = memnew(VisualShaderEditor);
 			shader_tabs->add_child(es.visual_shader_editor);
 			es.visual_shader_editor->edit(vs.ptr());
```

Comment:
This should likely be refactored further so that such hacks are not needed. A module should be modular and thus self-contained (sometimes it's not possible and we do allow hacks like this occasionally, but this needs to be assessed).

Reply:
Basically, if we want VisualShader to be a module, it should have the same relationship to the ShaderEditor and (script) Shader as VisualScript has to the ScriptEditor and GDScript.

There are no uses of `MODULE_VISUAL_SCRIPT_ENABLED` anywhere in the editor code, everything is self-contained. There are a few references by String though which are allowed:
```
$ rg '"VisualScript"'
editor_node.cpp
2306:                   if (main_plugin->get_name() == "Script" && !current_obj->is_class("VisualScript") && current_res && !current_res->is_built_in() && (bool(EditorSettings::get_singleton()->get("text_editor/external/use_external_editor")) || overrides_external_editor(current_obj))) {

editor_asset_installer.cpp
122:            extension_guess["vs"] = tree->get_theme_icon(SNAME("VisualScript"), SNAME("EditorIcons"));

translations/de.po
17338:msgstr "VisualScript"

plugins/script_editor_plugin.cpp
2270:                   !p_resource->is_class("VisualScript")) {
2369:   if (!p_resource->is_class("VisualScript")) {
```